### ISSUE TITLE
Error when installing plugin in TeamCity

### ISSUE THREAD

**Issue by WeyensLuk** [JetBrains/teamcity-dotnet-plugin#56]
Hi,

I'm trying to setup a .NET Core project in TeamCity and came across this plugin to facilitate my work in configuring a build. So I downloaded the plugin, copied it into the TeamCity datadirectory/plugins, rebooted the server and it booted with the following error:

```
[2017-09-05 13:10:42,960]  ERROR - gins.spring.SpringPluginLoader - Failed to initialize spring context for plugin dotnetcore 
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cacheManager' defined in class path resource [jetbrains/buildServer/dotnet/CachingConfig.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.cache.CacheManager]: Factory method 'cacheManager' threw exception; nested exception is java.lang.LinkageError: loader constraint violation: when resolving method "org.springframework.cache.guava.GuavaCacheManager.setCacheBuilder(Lcom/google/common/cache/CacheBuilder;)V" the class loader (instance of jetbrains/buildServer/plugins/classLoaders/PluginStandaloneClassLoader) of the current class, jetbrains/buildServer/dotnet/CachingConfig, and the class loader (instance of org/apache/catalina/loader/WebappClassLoader) for the method's defining class, org/springframework/cache/guava/GuavaCacheManager, have different Class objects for the type com/google/common/cache/CacheBuilder used in the signature
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:599)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1123)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1018)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:510)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:482)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:197)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:772)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:839)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:538)
	at jetbrains.buildServer.plugins.spring.SpringPluginLoader.pluginClassesLoaded(SpringPluginLoader.java:102)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at jetbrains.buildServer.util.EventDispatcher.dispatch(EventDispatcher.java:121)
	at jetbrains.buildServer.util.EventDispatcher$2.invoke(EventDispatcher.java:68)
	at com.sun.proxy.$Proxy16.pluginClassesLoaded(Unknown Source)
	at jetbrains.buildServer.plugins.PluginManagerImpl$2.visitPlugin(PluginManagerImpl.java:140)
	at jetbrains.buildServer.plugins.PluginsCollection$6.run(PluginsCollection.java:257)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.cache.CacheManager]: Factory method 'cacheManager' threw exception; nested exception is java.lang.LinkageError: loader constraint violation: when resolving method "org.springframework.cache.guava.GuavaCacheManager.setCacheBuilder(Lcom/google/common/cache/CacheBuilder;)V" the class loader (instance of jetbrains/buildServer/plugins/classLoaders/PluginStandaloneClassLoader) of the current class, jetbrains/buildServer/dotnet/CachingConfig, and the class loader (instance of org/apache/catalina/loader/WebappClassLoader) for the method's defining class, org/springframework/cache/guava/GuavaCacheManager, have different Class objects for the type com/google/common/cache/CacheBuilder used in the signature
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:189)
	at jetbrains.buildServer.spring.InstantiationStrategySelector$1.instantiate(InstantiationStrategySelector.java:90)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:588)
	... 26 more
Caused by: java.lang.LinkageError: loader constraint violation: when resolving method "org.springframework.cache.guava.GuavaCacheManager.setCacheBuilder(Lcom/google/common/cache/CacheBuilder;)V" the class loader (instance of jetbrains/buildServer/plugins/classLoaders/PluginStandaloneClassLoader) of the current class, jetbrains/buildServer/dotnet/CachingConfig, and the class loader (instance of org/apache/catalina/loader/WebappClassLoader) for the method's defining class, org/springframework/cache/guava/GuavaCacheManager, have different Class objects for the type com/google/common/cache/CacheBuilder used in the signature
	at jetbrains.buildServer.dotnet.CachingConfig.cacheManager(CachingConfig.kt:17)
	at jetbrains.buildServer.dotnet.CachingConfig$$EnhancerBySpringCGLIB$$44808dee.CGLIB$cacheManager$0(<generated>)
	at jetbrains.buildServer.dotnet.CachingConfig$$EnhancerBySpringCGLIB$$44808dee$$FastClassBySpringCGLIB$$f6a0cda1.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:228)
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:356)
	at jetbrains.buildServer.dotnet.CachingConfig$$EnhancerBySpringCGLIB$$44808dee.cacheManager(<generated>)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:162)
	... 28 more
```

Both server and its agents are running an embedded version of Java
```
java version "1.8.0_121"
Java(TM) SE Runtime Environment (build 1.8.0_121-b13)
Java HotSpot(TM) Client VM (build 25.121-b13, mixed mode)
```

Is there anything I'm missing? I have the dotnet CLI tools installed and didn't find any other prerequisites to get this working.

Thanks in advance,
Luk

**Comment by dtretyakov** [comment]
@WeyensLuk, could you please also report TeamCity and this plugin version?

**Comment by dtretyakov** [comment]
It seems that it was fixed as #54

**Comment by WeyensLuk** [comment]
True, I use TeamCity 2017.1.3, so it has indeed already been resolved, sorry for the double issue :)
